common/objectcachev2.h

```diff
@@ -164,20 +153,16 @@ class ObjectCacheV2 {
 
         Borrow() : _reader(nullptr) {}
 
-        Borrow(ObjectCacheV2* oc, Box* item, std::shared_ptr<V>&& reader)
-            : _oc(oc),
-              _item(item),
-              _reader(std::move(reader)),
-              _recycle(false) {}
+        Borrow(ObjectCacheV2* oc, Box* item, const std::shared_ptr<V>& reader)
+            : _oc(oc), _item(item), _reader(reader), _recycle(false) {}
 
         Borrow(Borrow&& rhs) : _reader(nullptr) { *this = std::move(rhs); }
 
         Borrow& operator=(Borrow&& rhs) {
```

Comment:
Instead of swap/exchange, it's more semantically right to:
```cpp
assert(_oc == rhs._oc);
std::atomic_exchange(&_reader, std::move(rhs._reader));
_item = rhs._item;
_recycle = rhs._recycle;
rhs._reader.reset();
rhs._item = nullptr;
rhs._recycle = false;
```

Reply:
according to cppreference https://en.cppreference.com/w/cpp/memory/shared_ptr

>  If multiple threads of execution access the same shared_ptr object without synchronization and any of those accesses uses a non-const member function of shared_ptr then a data race will occur

Since atomic<shared_ptr> is not supported by c++11/14/17, an new implementation using photon::mutex helps keep shared_ptr update in critical zone.